Refactor m_offset to align with Webkit styling in Position.cpp

Ahmad-S792 opened the pull request:
#### 5b848d9183ab8c666ae5ca856b5c753fa67f3d70
<pre>
Refactor m_offset to align with Webkit styling in Position.cpp

Refactor m_offset to align with Webkit styling in Position.cpp
<a href="https://bugs.webkit.org/show_bug.cgi?id=249090">https://bugs.webkit.org/show_bug.cgi?id=249090</a>

Reviewed by Ryosuke Niwa.

This patch is to refactor code around &apos;negative offset&apos; by changing it to Webkit Styling
to use negation (!) rather than (==) or (&gt;) etc.

* Source/WebCore/dom/Position.cpp:
(Position::parentAnchorEquivalent): Update m_offset
(Position::atFirstEditingPositionForNode): Ditto
(Position::atStartOfTree): Ditto

Canonical link: <a href="https://commits.webkit.org/257717@main">https://commits.webkit.org/257717@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9bde89e8e664e6204da9e638e954ef8669b5e28f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/99758 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/8933 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/32840 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/109122 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/169363 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/103761 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/9736 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/86226 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/92219 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/107031 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/105526 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/7367 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/90690 "Passed tests") | [💥 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/34142 "An unexpected error occured. Recent messages:Cleaned up git repository; Skipping applying patch since patch_id isn't provided; Checked out pull request; Updated gtk dependencies; Pull request contains relevant changes; Skipped layout-tests; layout-tests (exception)") | 
| | [⏳ 🧪 api-ios](https://ews-build.webkit.org/#/builders/API-Tests-iOS-Simulator-EWS "Waiting to run tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/22069 "Passed tests") | [⏳ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/API-Tests-GTK-EWS "Waiting to run tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/2755 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/23581 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/2701 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/45962 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5299 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/8836 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/43051 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/4561 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WebCore/dom/Position.cpp:
```diff
@@ -142,6 +143,7 @@ Position::Position(Text* textNode, unsigned offset)
     , m_anchorType(PositionIsOffsetInAnchor)
     , m_isLegacyEditingPosition(false)
 {
+    ASSERT(offset >= 0);
```
I don't think this makes much sense. offset is already unsigned here so it's literally impossible for it to be negative.

rniwa replied:
I guess this was changed in https://github.com/WebKit/WebKit/commit/6bd1d053f70a0dc5dbd3be29f331ee98e647e94b

Ahmad-S792 replied:
Should I remove this and keep remaining patch?

rniwa replied:
Yes. Removing assertion in other places makes sense because it's literally impossible for m_offset to be negative after  https://github.com/WebKit/WebKit/commit/6bd1d053f70a0dc5dbd3be29f331ee98e647e94b

Ahmad-S792 replied:
Done.

rniwa commented on Source/WebCore/dom/Position.cpp:
```diff
@@ -342,9 +344,6 @@ Position Position::previous(PositionMoveType moveType) const
         return *this;
 
-    // FIXME: Negative offsets shouldn't be allowed. We should catch this earlier.
-    ASSERT(deprecatedEditingOffset() >= 0);
```
Actually, this function still returns int so we should be asserting that it's non-negative.

Ahmad-S792 replied:
So I should close this PR, since we need these assertions still? Or we can add Assert at top in L146 and remove these?

rniwa replied:
No, I don't think we should remove these assertions or add that assertion.
Instead, the rest of code changes to use `!m_offset` instead of `m_offset > 0` is good.
We need to rename the PR / bug title accordingly though.

Ahmad-S792 replied:
Done.

rniwa commented on Source/WebCore/dom/Position.cpp:
```diff
@@ -396,9 +395,6 @@ Position Position::next(PositionMoveType moveType) const
         return *this;
 
-    // FIXME: Negative offsets shouldn't be allowed. We should catch this earlier.
-    ASSERT(deprecatedEditingOffset() >= 0);
```
Ditto.

Ahmad-S792 replied:
Done.

rniwa reviewed (approved):
r=me

rniwa reviewed (approved):
r=me

Ahmad-S792 committed:
Refactor m_offset to align with Webkit styling in Position.cpp

Refactor m_offset to align with Webkit styling in Position.cpp
https://bugs.webkit.org/show_bug.cgi?id=249090

Reviewed by Ryosuke Niwa.

This patch is to refactor code around 'negative offset' by changing it to Webkit Styling
to use negation (!) rather than (==) or (>) etc.

* Source/WebCore/dom/Position.cpp:
(Position::parentAnchorEquivalent): Update m_offset
(Position::atFirstEditingPositionForNode): Ditto
(Position::atStartOfTree): Ditto

Canonical link: https://commits.webkit.org/257717@main
